packages/react-devtools-extensions/src/parseHookNames.js

```diff
@@ -0,0 +1,501 @@
+            }
+          }
+
+          hookSourceData.sourceMapURL = url;
+
+          if (__DEBUG__) {
+            console.log(
+              'extractAndLoadSourceMaps() External source map "' + url + '"',
+            );
+          }
+
+          promises.push(
+            fetchFile(url).then(sourceMapContents => {
+              hookSourceData.sourceMapContents = JSON.parse(sourceMapContents);
+            }),
+          );
+          break;
+        }
+      }
+    }
+  });
+  return Promise.all(promises);
+}
+
+function fetchFile(url: string): Promise<string> {
+  return new Promise((resolve, reject) => {
+    fetch(url).then(response => {
+      if (response.ok) {
+        response
+          .text()
+          .then(text => {
+            resolve(text);
+          })
+          .catch(error => {
+            reject(null);
+          });
+      } else {
+        reject(null);
+      }
+    });
+  });
+}
+
+function findHookNames(
+  hooksList: Array<HooksNode>,
+  fileNameToHookSourceData: Map<string, HookSourceData>,
+): HookNames {
+  const map: HookNames = new Map();
+
+  hooksList.map(hook => {
+    if (isNonDeclarativePrimitiveHook(hook)) {
+      if (__DEBUG__) {
+        console.log('findHookNames() Non declarative primitive hook');
+      }
+
+      // Not all hooks have names (e.g. useEffect or useLayoutEffect)
+      return null;
+    }
+
+    // We already guard against a null HookSource in parseHookNames()
+    const hookSource = ((hook.hookSource: any): HookSource);
+    const fileName = hookSource.fileName;
+    if (!fileName) {
+      return null; // Should not be reachable.
+    }
+
+    const hookSourceData = fileNameToHookSourceData.get(fileName);
+    if (!hookSourceData) {
+      return null; // Should not be reachable.
+    }
+
+    const {lineNumber, columnNumber} = hookSource;
+    if (!lineNumber || !columnNumber) {
+      return null; // Should not be reachable.
+    }
+
+    const sourceConsumer = hookSourceData.sourceConsumer;
+
+    let originalSourceLineNumber;
+    if (areSourceMapsAppliedToErrors() || !sourceConsumer) {
+      // Either the current environment automatically applies source maps to errors,
+      // or the current code had no source map to begin with.
+      // Either way, we don't need to convert the Error stack frame locations.
+      originalSourceLineNumber = lineNumber;
+    } else {
+      originalSourceLineNumber = sourceConsumer.originalPositionFor({
+        line: lineNumber,
+        column: columnNumber,
+      }).line;
+    }
+
+    if (__DEBUG__) {
+      console.log(
+        'findHookNames() mapped line number',
+        lineNumber,
+        'to',
+        originalSourceLineNumber,
+      );
+    }
+
+    if (originalSourceLineNumber === null) {
+      return null;
+    }
+
+    const name = getHookName(
+      hook,
+      hookSourceData.originalSourceAST,
+      ((hookSourceData.originalSourceCode: any): string),
+      ((originalSourceLineNumber: any): number),
```

Comment:
Are we assuming that a line number is sufficient to identify a hook call? Seems like that assumption breaks down in edge cases:

1. Minified code (either without a source map, or a bundle *built* from pre-minified inputs) can have multiple hooks on one line.
2. Some code might be authored with quirky formatting, i.e. not with a typical Prettier / ESLint setup.

IMO it's safer to use all the information we have from the source map / AST and consider both lines and columns everywhere.

Reply:
> IMO it's safer to use all the information we have from the source map / AST and consider both lines and columns everywhere.

Yeah. Sounds reasonable. I've created #21792 to follow up.